`polars.DataFrame.write_excel` parameter `hidden_columns` is only used if `column_widths` is also set

### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import io
import polars as pl


df = pl.DataFrame(
    {
        "a": [1, 2, 3, 4, 5],
        "b": [1, 2, 3, 4, 5],
    }
)
buf_hidden = io.BytesIO()
buf_not_hidden = io.BytesIO()
df.write_excel(
    buf_hidden,
    hidden_columns=["a"],
)
df.write_excel(
    buf_not_hidden,
)
assert buf_not_hidden.getvalue() != buf_hidden.getvalue(), "should not be equal"
```

### Log output

```
{
	"name": "AssertionError",
	"message": "should not be equal",
	"stack": "---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[3], line 21
     14 df.write_excel(
     15     buf_hidden,
     16     hidden_columns=[\"a\"],
     17 )
     18 df.write_excel(
     19     buf_not_hidden,
     20 )
---> 21 assert buf_not_hidden.getvalue() != buf_hidden.getvalue(), \"should not be equal\"

AssertionError: should not be equal"
}
```

### Issue description

`hidden_columns` is only respected if the column is also affected by `column_widths`.

You can easily see the problem in the source code:
https://github.com/pola-rs/polars/blob/71a8b0543362aadb771c2a446ac93756ea36a124/py-polars/polars/dataframe/frame.py#L3453-L3464

`option` is not used anywhere else.

It works, if you adapt it to:

```python
        for column in df.columns:
            col_idx, options = table_start[1] + df.get_column_index(column), {}
            if column in hidden_columns:
                options = {"hidden": True}
            if column in column_widths:  # type: ignore[operator]
                ws.set_column_pixels(
                    col_idx,
                    col_idx,
                    column_widths[column],  # type: ignore[index]
                    None,
                    options,
                )
            elif options:
                ws.set_column(
                    col_idx,
                    col_idx,
                    None,
                    None,
                    options,
                )
```

I currently don't have the time to setup the project and investigate how to provide a proper test case, so feel free to use the snippet to correct the function.

### Expected behavior

Respect `hidden_columns` indipendently from `column_widths`.

### Installed versions

<details>

```
--------Version info---------
Polars:              1.8.2
Index type:          UInt32
Platform:            Linux-6.3.1-arch1-1-x86_64-with-glibc2.37
Python:              3.12.3 (main, Apr 22 2024, 13:23:18) [GCC 13.1.1 20230429]

----Optional dependencies----
adbc_driver_manager  1.2.0
altair               <not installed>
cloudpickle          3.0.0
connectorx           0.3.3
deltalake            <not installed>
fastexcel            0.11.6
fsspec               2024.9.0
gevent               <not installed>
great_tables         0.11.1
matplotlib           3.9.2
nest_asyncio         1.6.0
numpy                1.26.4
openpyxl             3.1.5
pandas               2.2.3
pyarrow              17.0.0
pydantic             2.9.2
pyiceberg            <not installed>
sqlalchemy           2.0.35
torch                2.4.1+cu121
xlsx2csv             0.8.3
xlsxwriter           3.2.0
```

</details>

--- a/py-polars/polars/dataframe/frame.py
+++ b/py-polars/polars/dataframe/frame.py
@@ -3121,8 +3121,9 @@ class DataFrame:
             If the table has headers, provide autofilter capability.
         autofit : bool
             Calculate individual column widths from the data.
-        hidden_columns : list
-             A list or selector representing table columns to hide in the worksheet.
+        hidden_columns : str | list
+             A column name, list of column names, or a selector representing table
+             columns to mark as hidden in the output worksheet.
         hide_gridlines : bool
             Do not display any gridlines on the output worksheet.
         sheet_zoom : int
@@ -3442,10 +3443,15 @@ class DataFrame:
                     include_header=include_header,
                     format_cache=fmt_cache,
                 )
+
         # additional column-level properties
         if hidden_columns is None:
-            hidden_columns = ()
-        hidden_columns = _expand_selectors(df, hidden_columns)
+            hidden = set()
+        elif isinstance(hidden_columns, str):
+            hidden = {hidden_columns}
+        else:
+            hidden = set(_expand_selectors(df, hidden_columns))
+
         if isinstance(column_widths, int):
             column_widths = dict.fromkeys(df.columns, column_widths)
         else:
@@ -3455,9 +3461,8 @@ class DataFrame:
         column_widths = _unpack_multi_column_dict(column_widths or {})  # type: ignore[assignment]
 
         for column in df.columns:
-            col_idx, options = table_start[1] + df.get_column_index(column), {}
-            if column in hidden_columns:
-                options = {"hidden": True}
+            options = {"hidden": True} if column in hidden else {}
+            col_idx = table_start[1] + df.get_column_index(column)
             if column in column_widths:  # type: ignore[operator]
                 ws.set_column_pixels(
                     col_idx,
@@ -3466,6 +3471,8 @@ class DataFrame:
                     None,
                     options,
                 )
+            elif options:
+                ws.set_column(col_idx, col_idx, None, None, options)
 
         # finally, inject any sparklines into the table
         for column, params in (sparklines or {}).items():

